common/src/main/java/org/terracotta/angela/common/distribution/WatchedProcess.java

```diff
@@ -30,24 +31,24 @@
   private final int pid;
 
   public WatchedProcess(ProcessExecutor processExecutor, final AtomicReference<S> stateRef, final S deadState) {
+    processExecutor.addListener(new ProcessListener() {
+      @Override
+      public void afterFinish(Process process, ProcessResult result) {
+        stateRef.set(deadState);
+      }
+
+      @Override
+      public void afterStop(Process process) {
+        stateRef.set(deadState);
+      }
+    });
+
```

Comment:
Use the ZT listener mechanism instead to record STOPPED (dead) state:
- Invoked after a process has finished successfully.
- Invoked after a process has exited (whether finished or cancelled).